### PR TITLE
Use vkCmdWriteTimestamp2 instead

### PR DESCRIPTION
**Pull request by libreliu**
Hi,

Thanks for the awesome library! But a minor issue exists to prevent this library building out of the box.

This commit fixes the build problem under VS2019 with Vulkan 1.3.236.0 installed. The KHR singatured version will actually break when using Auto-Vk in its default settings, that is, statitically linking with the core functions while using dynamically linking with the extension functions.

Though Auto-Vk-Toolkit is not affected since it uses dynamically linking for all of its functions, I'm putting this fix here if anyone interested in using this library alone.

Please let me know if this fix is appropriate.

Thanks!

### CODE REVIEW

**Comment by johannesugb** [comment]
Thank you for this PR! I'll merge after doing a quick test in a few days.

**Review by johannesugb — COMMENTED** [review]
Yeah, indeed this is in bad form. ^^ 
Could you try if the changes I pushed to the `vkTimestamp2`-branch work for you? => https://github.com/cg-tuwien/Auto-Vk/tree/writeTimestamp2

Gotta still come up with a better preprocessor check though, before I merge.

**Comment by libreliu** [comment]
Sorry, haven't checking this for a moment...
Judging from the thread, it seems a little tricky to support for both Vulkan 1.2 and 1.3..?
I'm wondering if it's okay to always query for VK_KHR_synchronization2 in Vulkan 1.2, and use the KHR suffixed version for both 1.2 and 1.3, since the spec says "The original type, enum and command names are still available as aliases of the core functionality.".

File: src/avk.cpp
Comment (johannesugb) [inline src/avk.cpp:7539]:
```suggestion
				cb.handle().writeTimestamp2KHR(lTimestampStage, lHandle, aQueryIndex, cb.root_ptr()->dispatch_loader_ext());

```

**Review by johannesugb — CHANGES_REQUESTED** [review]
After giving some more thought to it, I think the best way to go for now is to stick to enabling Synchronization2 as a Vulkan 1.2 extension via [`VkPhysicalDeviceSynchronization2Features`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceSynchronization2Features.html) and as a consequence, sticking to the matching `writeTimestamp2KHR` (i.e. **KHR**) function. 

The problem with the code here is, that `dispatch_loader_core` has been used, but the way how Synchronization2 is used is through an extension. Therefore, `dispatch_loader_ext` would be appropriate. 

May I ask you to check if you application runs fine when using `writeTimestamp2KHR` with `dispatch_loader_ext` as in the suggested change?

**Comment by Black-Photon** [comment]
Not the original author, but I came across this issue as well when getting started with the library, and the proposed change worked for me.

**Comment by libreliu** [comment]
Thoughts: 
It seems way more convenient for vulkan loaders to be linked dynamically in terms of programming, since all the nasty KHR promotion issues will be solved (will it be?) or at least get postponed to where a program actually hits a Vulkan 1.2 implementation. (I'm not sure if a 1.3 loader will "patch" those promoted calls back to 1.2's KHR suffixed version when using a 1.2-compliant vulkan icd driver)

So what are the real advantages for static linking the Vulkan Loader? Do they get a performance boost for something like inlining calls to the loaders..? (Less likely? since the final driver endpoint could be cached at the user-side?)

**Comment by johannesugb** [comment]
This issue should be resolved with Pull Request #83 
There's a new preprocessor define `AVK_USE_CORE_INSTEAD_OF_SYNCHRONIZATION2` which can be used to switch between Synchronization2 and core.

In addition to that, the defines `DISPATCH_LOADER_CORE_TYPE`, `DISPATCH_LOADER_EXT_TYPE`, and `VULKAN_HPP_DISPATCH_LOADER_DYNAMIC` can be used to configure the behavior of Auto-Vk.

**Pull request closed without merging**
